fhir-persistence-cassandra/src/main/java/com/ibm/fhir/persistence/cassandra/cql/DatasourceSessions.java

```diff
@@ -0,0 +1,210 @@
+     * schema creation where we create the keyspace for the first time
+     * @return
+     */
+    public static CqlSession getSessionForBootstrap(String tenantId, String dsId) {
+
+        String dsPropertyName = FHIRConfiguration.PROPERTY_DATASOURCES + "/" + dsId;
+        PropertyGroup dsPG = FHIRConfigHelper.getPropertyGroup(dsPropertyName);
+        if (dsPG == null) {
+            throw new IllegalStateException("Could not locate configuration property group: " + dsPropertyName);
+        }
+
+        try {
+            // Get the datasource type (should be cassandra in this case).
+            String type = dsPG.getStringProperty("type", null);
+            if (type == null) {
+                throw new IllegalStateException("Could not locate 'type' property within datasource property group: " + dsPropertyName);
+            }
+    
+            // Confirm that this is a cassandra datasource configuration element
+            if (!"cassandra".equals(type)) {
+                throw new IllegalStateException("Unsupported 'type' property value within datasource property group: " + type);  
+            }
+    
+            // Get the connection properties
+            PropertyGroup connectionProps = dsPG.getPropertyGroup("connectionProperties");
+            if (connectionProps == null) {
+                throw new IllegalStateException("Could not locate 'connectionProperties' property group within datasource property group: " + dsPropertyName);
+            }
+            
+            TenantDatasourceKey key = new TenantDatasourceKey(tenantId, dsId);
+            CassandraPropertyGroupAdapter adapter = new CassandraPropertyGroupAdapter(connectionProps);
+            return getDatabaseSession(key, adapter, false);
+        }
+        catch (Exception x) {
+            throw new IllegalStateException(x);
+        }
+    }
+
+    /**
+     * Get rid of any sessions we're holding on to
+     */
+    private void closeAllSessions() {
+        // prevent anyone asking for a session
+        this.running = false;
+        sessionMap.forEach((k,s) -> closeSession(k, s));
+        sessionMap.clear();
+    }
+
+    /**
+     * Safely close the session associated with this {@link TenantDatasourceKey}, catching
+     * and logging any exceptions. Does not rethrow any exception caught when trying to close the
+     * session.
+     * @param key
+     * @param s
+     */
+    private void closeSession(TenantDatasourceKey key, CqlSession s) {
+        try {
+            logger.info("Closing session for tenant/datasource: '"
+                    + key.getTenantId() + "/" + key.getDatasourceId() + "'");
+            s.close();
+        } catch (Exception x) {
+            logger.log(Level.WARNING, "failed to close session for tenant/datasource: '" 
+                    + key.getTenantId() + "/" + key.getDatasourceId() + "'", x);
+        }
+    }
+
+    /**
+     * 
+     */
+    public static void shutdown() {
```

Comment:
When is this called? Is this hooked on the servlet context listener? What's the purpose? Run on the commandline?  A comment would help